compositor_pipeline/src/pipeline/output/mp4.rs

```diff
@@ -269,18 +271,38 @@ fn create_packet(
         }
     }?;
 
+    // Starting output PTS from 0
+    let timestamp_offset = stream_state.timestamp_offset(&chunk);
+    let pts = chunk.pts.saturating_sub(timestamp_offset);
+    let dts = chunk
+        .dts
+        .map(|dts| dts.saturating_sub(timestamp_offset))
+        .unwrap_or(pts);
+
     let mut packet = ffmpeg::Packet::copy(&chunk.data);
-    packet.set_pts(Some((chunk.pts.as_secs_f64() * timebase) as i64));
-    let dts = chunk.dts.unwrap_or(chunk.pts);
-    packet.set_dts(Some((dts.as_secs_f64() * timebase) as i64));
-    packet.set_time_base(ffmpeg::Rational::new(1, timebase as i32));
-    packet.set_stream(stream_id);
+    packet.set_pts(Some((pts.as_secs_f64() * stream_state.time_base) as i64));
+    packet.set_dts(Some((dts.as_secs_f64() * stream_state.time_base) as i64));
+    packet.set_time_base(ffmpeg::Rational::new(1, stream_state.time_base as i32));
+    packet.set_stream(stream_state.id);
 
     Some(packet)
 }
 
 #[derive(Debug, Clone)]
-struct Stream {
+struct StreamState {
     id: usize,
     time_base: f64,
+    timestamp_offset: Option<Duration>,
+}
+
+impl StreamState {
+    fn timestamp_offset(&mut self, chunk: &EncodedChunk) -> Duration {
+        match self.timestamp_offset {
```

Comment:
I think that my suggestion to use `get_or_insert_with` still would make sense here